component-tools/src/main/java/org/talend/sdk/component/tools/ComponentValidator.java

```diff
@@ -156,23 +160,52 @@ private String validateIcon(final Icon annotation, final Collection<String> erro
 
         if (annotation.value() == Icon.IconType.CUSTOM) {
             final String icon = annotation.custom();
-            final Set<File> svgs =
-                    of(classes).map(it -> new File(it, "icons/" + icon + ".svg")).filter(File::exists).collect(toSet());
-            if (svgs.isEmpty()) {
-                log.error("No 'icons/" + icon + ".svg' found, this will run in degraded mode in Talend Cloud");
+            Set<File> svgs;
+            Set<File> pngs;
+            // legacy checks
+            if (configuration.isValidateLegacyIcons()) {
+                svgs = of(classes)
+                        .map(it -> new File(it, ICONS + icon + ".svg"))
+                        .collect(toSet());
+                pngs = Stream.of(classes)
+                        .map(it -> new File(it, ICONS + icon + "_icon32.png"))
+                        .collect(Collectors.toSet());
             } else {
-                if (configuration.isValidateSvg()) {
-                    errors.addAll(svgs.stream().flatMap(this::validateSvg).collect(toSet()));
-                }
+                // themed icons check
+                List<String> prefixes = new ArrayList<>();
+                of(classes).forEach(s -> {
+                    prefixes.add(s + "/icons/light/" + icon);
```

Comment:
and here it's hardcoded, but the the level of the constant is the same

Reply:
hum, yay can  introduce the constant.